Documentation/CRDs/Cluster/external-cluster/topology-for-external-mode.md

```diff
@@ -0,0 +1,199 @@
+# Topology provisioning for external mode
+
+## Background
+For Block storage (RWO), Enable Topology awareness for PV provisioning. Preferably, picking the replica 1 Storage Class for this AZ or creating a volume in a relevant AZ using a single storage class. It will help in providing Isolation to application/pods in AZ
+
+## Scenario
+Applications like Kafka will have a deployment with multiple running instances. Each service instance will create a new claim and is expected to be located in a different zone. Since the application has its own redundant instances, there is no requirement for redundancy at the data layer. A storage class is created that will provision storage from replica 1 Ceph pools that are located in each of the separate zones.
+
+## Technical implementation details
+
+Add --topology-pools, --topology-failure-domain-label, --topology-failure-domain-values to the script: `create-external-cluster-resources.py` where,
+`--topology-pools`: (optional) comma-separated list of topology-constrained rbd pools
+`--topology-failure-domain-label`: (optional) k8s cluster failure domain label (example: zone,rack,host,etc) for the topology-pools that are matching the ceph domain
+ `--topology-failure-domain-values`: (optional) comma-separated list of the k8s cluster failure domain values corresponding to each of the pools in the topology-pools list
+
+The import script will then create a new storage class named `ceph-rbd-topology`.
+
+## Example Configuration
+
+### Ceph cluster
+
+Looking at the osd tree, before jumping to the steps
+```
+ID  CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
+-1         2.00000  root default
+-9         0.50000      host osd-0
+ 3    hdd  0.25000          osd.3       up   1.00000  1.00000
+ 7    hdd  0.25000          osd.7       up   1.00000  1.00000
+-7         0.50000      host osd-1
+ 1    hdd  0.25000          osd.1       up   1.00000  1.00000
+ 6    hdd  0.25000          osd.6       up   1.00000  1.00000
+-3         0.50000      host osd-2
+ 2    hdd  0.25000          osd.2       up   1.00000  1.00000
+ 5    hdd  0.25000          osd.5       up   1.00000  1.00000
+-5         0.50000      host osd-3
+ 0    hdd  0.25000          osd.0       up   1.00000  1.00000
+ 4    hdd  0.25000          osd.4       up   1.00000  1.00000
+```
+
+Create m replica-n pools, in the example create 3 replica-1 pools:
+
+```
+pool 10 'ab' replicated size 1 min_size 1 crush_rule 5 object_hash rjenkins pg_num 32 pgp_num 32 autoscale_mode on last_change 334 lfor 0/0/254 flags hashpspool,selfmanaged_snaps stripe_width 0 application rbd read_balance_score 1.06
```

Comment:
IMO no need to show this verbose pool output. You could suggest they run `ceph osd pool ls detail` to verify the pools are replica 1, but no need for all this output.